docs/kernel_finder.rst

```diff
@@ -0,0 +1,183 @@
+.. _kernel_finder:
+
+================
+Kernel Discovery
+================
+
+The primary purpose of the Jupyter Kernel management package is to provide
+a means of discovering kernels that are available for use.  This is accomplished
+using the :class:`KernelFinder <jupyter_kernel_mgmt.discovery.KernelFinder>` class.
+
+:class:`KernelFinder <jupyter_kernel_mgmt.discovery.KernelFinder>` instances are created in one of two ways.
+
+1. The most common way is to call KernelFinder's class method 
+:meth:`.KernelFinder.from_entrypoints()`. This loads all of the registered
+kernel providers.
+
+2. You can also provide a list of :class:`KernelProvider <jupyter_kernel_mgmt.discovery.KernelProviderBase>`
+instances via KernelFinder's initializer: :meth:`KernelFinder(providers) <jupyter_kernel_mgmt.discovery.KernelFinder>`.
+This loads only those instances provided.
+
+Once an instance of :class:`KernelFinder <jupyter_kernel_mgmt.discovery.KernelFinder>` has
+been created, kernels can be discovered and launched via KernelFinder's
+instance methods, :meth:`find_kernels() <.KernelFinder.find_kernels>` and
+:meth:`launch() <.KernelFinder.launch>`, respectively.
+
+Finding kernels
+===============
+
+The set of currently available kernel types are discovered using KernelFinder's
+:meth:`.KernelFinder.find_kernels()` method.  This method is a generator that walks
+the set of loaded kernel providers calling each of their
+:meth:`KernelProvider.find_kernels() <.KernelProviderBase.find_kernels()>` methods
+yielding each entry.
+
+Each entry, commonly referred to as a :ref:`kernel specification <kernelspecs>`, is a JSON serialized
+dictionary consisting of the fields necessary to describe and use the associated kernel.  Some
```

Comment:
What you get from `find_kernels()` is not JSON serialised. It should be restricted to data that *could* be JSON serialised, but it gives you dicts, not strings.